pkg/pubsub/dapr/dapr.go

```diff
@@ -0,0 +1,100 @@
+package dapr
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/dapr/go-sdk/client"
+	"github.com/open-policy-agent/gatekeeper/pkg/pubsub/common"
+	"github.com/open-policy-agent/gatekeeper/pkg/pubsub/connection"
+)
+
+type ClientConfig struct {
+	// Enable, Disable batching per tool
+	EnableBatching bool `json:"enableBatching,omitempty"`
+
+	// batch size
+	Size int `json:"size"`
+
+	// Name of the component to be used for pub sub messaging
+	Component string `json:"component"`
+}
+
+// Dapr represents driver for interacting with pub sub using dapr.
+type Dapr struct {
+	// Array of clients to talk to different endpoints
+	client client.Client
+
+	// Name of the pubsub component
+	pubSubComponent string
+
+	// Enable, Disable batching per tool
+	batchingEnabled bool
+
+	// batch size
+	size int
+}
+
+const (
+	Name          = "dapr"
+	componentName = "pubsub"
+)
+
+func (r *Dapr) Publish(ctx context.Context, data interface{}, topic string) error {
+	jsonData, err := json.Marshal(data)
+	if err != nil {
+		return fmt.Errorf("error marshaling data: %w", err)
+	}
+
+	err = r.client.PublishEvent(context.Background(), r.pubSubComponent, topic, jsonData)
+	if err != nil {
+		return fmt.Errorf("error publishing message to dapr: %w", err)
+	}
+
+	return nil
+}
+
+func (r *Dapr) PublishBatch(ctx context.Context, data interface{}, topic string) error {
+	// add logic of batch publishing with dapr from 1.10
+	return nil
+}
+
+// Determines if batching is enabled or not.
+func (r *Dapr) IsBatchingEnabled() bool {
+	return r.batchingEnabled
+}
+
+// Returns a new client for dapr.
+func NewConnection(ctx context.Context, config interface{}) (connection.Connection, error) {
+	var cfg ClientConfig
+	m, ok := config.(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("invalid type assertion, config is not in extected format")
+	}
+	cfg.EnableBatching, ok = m["enableBatching"].(bool)
+	if !ok {
+		return nil, fmt.Errorf("failed to get value of `enableBatching`")
```

Comment:
error messages are not markdown, no need for ``s

Reply:
Nice catch!